lib/wsdl.js

```diff
@@ -1206,7 +1206,98 @@ WSDL.prototype.toXML = function() {
+            }
+            var typeDef = this.findSchemaObject(typeURI, type.name);
+            if (typeDef) {
+                xsiTypeSchema = typeDef.description(this.definitions);
+            }
+        }
+    }
+    return xsiTypeSchema;
+};
+
+WSDL.prototype.findSchema = function (topSchema, name, originalName) {
+    var schema;
+    if (topSchema) {
+        // Find the current object's schema
+        if (topSchema[name + '[]'])
+            name = name + '[]';
+
+        if (topSchema.name) {
+            if (topSchema.name === 'complexType') {
+                schema = this.findSchema(topSchema.children[0], name, originalName);
+            } else if (topSchema.name === 'sequence') {
+                // Linear search among the elements of the sequence
+                for (var i = 0; i < topSchema.children.length; i++) {
+                    if (topSchema.children[i].$name === originalName) {
+                        schema = topSchema.children[i];
+                        break;
+                    }
+                }
+
+                if (schema && schema.name === 'element') {
+                    var splitPrefix_Name = splitQName(schema.$type);
+                    if (splitPrefix_Name.prefix !== 'xs') {
+                        schema = this.findInNamespace(splitPrefix_Name);
+                    } else {
+                        schema = schema.$type;
+                    }
+                }
+            } else if (topSchema.name === 'complexContent' && topSchema.children[0].name === 'restriction' && topSchema.children[0].$base === 'soapenc:Array') {
+                schema = topSchema.children[0].children[1];
+                if ('$ref' in schema && schema.$ref === 'soapenc:arrayType' && '$n1:arrayType' in schema) {
+                    schema = schema['$n1:arrayType'].substring(0, schema['$n1:arrayType'].length - 2);
+                    if (schema.substring(0, 2) !== 'xs') {
+                        var splitPrefix_Name = splitQName(schema);
+                        schema = this.findInNamespace(splitPrefix_Name);
+                    }
+                }
+            } else if (topSchema.name === 'complexContent' && topSchema.children[0].name === 'extension') {
+                schema = this.findSchema(topSchema.children[0].children[0], name, originalName);
+                if (!schema) {
+                    // Handle class inheritance recursively
+                    var splitPrefix_Name = splitQName(topSchema.children[0].$base);
+                    var baseSchema = this.findInNamespace(splitPrefix_Name);
+                    schema = this.findSchema(baseSchema, name, originalName);
+                }
+            } else if (typeof topSchema.xmlns === 'string') {
+                schema = this.definitions.schemas[topSchema.xmlns].complexTypes[originalName];
+            }
+        } else {
+            schema = (topSchema && topSchema[name]);
+        }
+    }
+    return(schema);
```

Comment:
Please remove the surrounding `()` here as they are not needed and are not "conventional" in our codebase